stub/src/main/java/io/grpc/stub/Calls.java

```diff
@@ -188,7 +190,7 @@
   public static <ReqT, RespT> RespT blockingClientStreamingCall(
       Call<ReqT, RespT> call,
       Iterator<ReqT> clientStream) {
-    SettableFuture<RespT> responseFuture = SettableFuture.create();
+    GrpcFuture<RespT> responseFuture = new GrpcFuture<RespT>(call);
```

Comment:
No it won't work. AbstractFuture#set() is protected. If UnaryStreamToFuture accepted AbstractFuture, it would not be able to call set().